### PR TITLE
Rearranged and edited basic cli ops info

### PR DESCRIPTION
**Pull request by bfallonf**
@vikram-redhat @adellape This is the PR about making the cli operations doc more fluid and without repeated info. I made it more like the `oc` command's output, and added examples, as well as pointed to the more info docs (eg, deployments, builds, templates, etc.) when appropriate.

What do you guys think?

### CODE REVIEW

File: cli_reference/basic_cli_operations.adoc [hunk 1]
```diff
@@ -230,158 +195,145 @@ operations for working with application builds:
...
-----
-
-The following example describes a deployment configuration called
-`docker-registry`:
+|Retrieve the build logs for the specified build.
 
-====
+|`deploy`
+|`oc deploy _<deploymentconfig>_`
+|View a link:../dev_guide/deployments.html[deplopyment], or manually start, cancel, or retry a deployment.
```
Comment (ironcladlou) [inline cli_reference/basic_cli_operations.adoc]: deplopyment -> deployment

File: dev_guide/deployments.adoc [hunk 2]
```diff
@@ -138,16 +170,28 @@ $ osc describe dc <deployment_config>
...
 ----
-$ oc deploy <deployment_config> --cancel
+$ oc deploy <config_name> --cancel
 ----
 
-WARNING: The cancelation is a best-effort operation, and may take some time to
-complete. It's possible the deployment will partially or totally complete
-before the cancelation is effective.
+After running this command, the deployment eventually shuts down and transitions
+to a Failed status.
```
Comment (ironcladlou) [inline dev_guide/deployments.adoc]: I don't think I like the "Failed status" part since deployment "states" or "statuses" aren't described anywhere, making this seem a bit out of context. Do you think the detail provided in my warning is irrelevant?

File: dev_guide/deployments.adoc [hunk 3]
```diff
@@ -109,6 +109,38 @@ created each time a new version of the `origin-ruby-sample:latest` image reposit
...
+
+[options="nowrap"]
+----
+$ oc describe dc docker-registry
+Name:		docker-registry
+Created:	18 hours ago
+Labels:		docker-registry=default
+Latest Version:	1
+Triggers:	Config
+Strategy:	Recreate
```
Comment (ironcladlou) [inline dev_guide/deployments.adoc]: Let's promote the default Rolling strategy here

File: dev_guide/deployments.adoc [hunk 4]
```diff
@@ -138,16 +170,28 @@ $ osc describe dc <deployment_config>
...
+[NOTE]
+====
+By default, the deployment process is restricted to 6 hours. If a deployment is
+still in progress after 6 hours, then the containers for the deployment pod and
+any deployment hook pods are all deleted, and these pods are placed in the
+Failed state.
+====
 
-To cancel a running deployment:
+Cancel a running or stuck deployment with the following command:
```
Comment (ironcladlou) [inline dev_guide/deployments.adoc]: More consistent with the other examples would be:

"To cancel a running or stuck deployment:"

File: dev_guide/deployments.adoc [hunk 5]
```diff
@@ -138,16 +170,28 @@ $ osc describe dc <deployment_config>
...
 
 ----
-$ oc deploy <deployment_config> --cancel
+$ oc deploy <config_name> --cancel
```
Comment (ironcladlou) [inline dev_guide/deployments.adoc]: All the other examples use deployment_config... I'd make them all uniform one way or the other.

File: dev_guide/deployments.adoc [hunk 6]
```diff
@@ -165,27 +209,78 @@ deployment version. The restarted deployment will have the same configuration
...
+|Template string or path to template file to use when `-o=template` or `-o=templatefile`.
+|===
+
+To perform a rollback:
+
+====
 
 ----
-$ oc rollback <deployment>
+$ oc rollback deployment-1
```
Comment (ironcladlou) [inline dev_guide/deployments.adoc]: All the other examples use a <resource_description> style... why the change here?

File: dev_guide/deployments.adoc [hunk 7]
```diff
@@ -165,27 +209,78 @@ deployment version. The restarted deployment will have the same configuration
...
+====
 
-The deployment configuration's template will be reverted to match the
-deployment specified in the rollback command, and a new deployment will be
-started.
+To see what the rollback will look like without performing the rollback:
 
-Image change triggers on the deployment configuration are disabled as part of
-the rollback to prevent unwanted deployments soon after the rollback is
-complete. To re-enable the image change triggers:
```
Comment (ironcladlou) [inline dev_guide/deployments.adoc]: This was important... why remove it?

File: dev_guide/deployments.adoc [hunk 8]
```diff
@@ -165,27 +209,78 @@ deployment version. The restarted deployment will have the same configuration
...
 
 ----
-$ oc deploy <deployment_config> --enable-triggers
+$ oc rollback deployment-1 --dry-run
 ----
+====
 
-NOTE: More fine grained rollback options are described in the
-link:../cli_reference/basic_cli_operations.html#deployment-operations[deployment operations reference].
+To perform the rollback manually by piping the *JSON* of the new configuration back to `oc`:
```
Comment (ironcladlou) [inline dev_guide/deployments.adoc]: In hindsight I think this example is more noisy than it is useful.

**Comment by bfallonf** [comment]
@ironcladlou I apologise. Most of the comments you made above were the results of a slap-dash copy/paste from me trying to get the docs to allign. Seems I copied over some of the previous stuff without paying the right amount of attention.

The good news is that after the reorganising of the architecture sections caused a load of conflicts with this stuff, your work was restored. The only outstanding comment was the one about the intro para to "Canceling a Deployment", so I've fixed that. Other than that, the rest of your stuff is still in there.

@vikram-redhat @adellape comment asking for thoughts on this still applies.

**Comment by adellape** [comment]
LGTM

**Comment by adellape** [comment]
:shipit:
